### PR TITLE
Cacao-zcap SDK support.

### PR DESCRIPTION
**Pull request by cobward**
Sister kepler-sdk PR: https://github.com/spruceid/kepler-sdk/pull/40

* Rewrite core SDK functionality in rust.
* Refactor out some core definitions from kepler to kepler-lib.
* Support cacao-zcaps in the SDK.

### CODE REVIEW

**Review by chunningham — APPROVED** [review]
overall LGTM, in future we can think about what else we can move into Kepler-lib, but this works well, is pretty clean and does all we need. is the typescript type declaration in `sdk-wasm::host` something we can extend to all fns/does it generate typed TS fns on the other side? I wouldn't think so but I live in hope

File: sdk-wasm/src/serde_siwe.rs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    {
+        Option::<String>::deserialize(d)?
+            .as_deref()
+            .map(TimeStamp::from_str)
+            .transpose()
+            .map_err(|e| D::Error::custom(format!("failed to parse timestamp: {}", e)))
+    }
+}
+
+pub mod timestamp {
```
[the remark this answers is not in the document]
Comment (cobward) [reply sdk-wasm/src/serde_siwe.rs:50]: TimeStamp is external to siwe-rs too I believe.

[the remark this answers is not in the document]
Comment (cobward) [reply sdk-wasm/src/serde_siwe.rs:50]: Could be behind a feature flag? Seems reasonable that someone might want to have Ser+De implemented for `Message`

File: lib/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+[package]
+name = "kepler-lib"
+version = "0.1.0"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+[features]
+"default" = ["didkit/ring"]
```
[the remark this answers is not in the document]
Comment (cobward) [reply lib/Cargo.toml:8]: Seems like there's no reason _to_ use ring. I'll try just using the wasm-compatible crypto for everything and make sure nothing breaks.

[the remark this answers is not in the document]
Comment (chunningham) [reply lib/Cargo.toml]: yeah honestly don't remember exactly why but I am in favour of one-basis-if-possible

**Pull request merged**